test/OpenTelemetry.Instrumentation.Owin.Tests/DiagnosticsMiddlewareTests.cs

```diff
@@ -222,5 +246,63 @@ public void Dispose()
         {
             Assert.Empty(stoppedActivities);
         }
+
+        if (instrumentMetrics)
+        {
+            meterProvider.Dispose();
+            var metric = exportedMetrics[0];
+            var metricPoints = this.GetMetricPoints(metric);
+            var metricPoint = Assert.Single(metricPoints);
+
+            Assert.Equal(OwinInstrumentationMetrics.MeterName, metric.MeterName);
+            Assert.Equal("http.server.request.duration", metric.Name);
+            Assert.Equal(MetricType.Histogram, metric.MetricType);
+            Assert.Equal(1, metricPoint.GetHistogramCount());
+            Assert.Equal(3, metricPoint.Tags.Count);
+
+            foreach (var tag in metricPoint.Tags)
+            {
+                switch (tag.Key)
+                {
+                    case SemanticConventions.AttributeHttpMethod:
+                        Assert.Equal("GET", tag.Value);
+                        break;
+                    case SemanticConventions.AttributeHttpScheme:
+                        Assert.Equal(requestUri.Scheme, tag.Value);
+                        break;
+                    case SemanticConventions.AttributeHttpStatusCode:
+                        Assert.Equal(generateRemoteException ? 500 : 200, tag.Value);
+                        break;
+                }
+            }
+        }
+        else
+        {
+            Assert.Empty(exportedMetrics);
+        }
+
+        if (instrumentMetrics && instrumentTraces && !filter)
+        {
+            var metric = Assert.Single(exportedMetrics);
+            var activity = Assert.Single(stoppedActivities);
+            var metricPoints = this.GetMetricPoints(metric);
+            var metricPoint = Assert.Single(metricPoints);
+
+            // metric value and span duration should match
+            // https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md#metric-httpserverrequestduration
+            Assert.Equal(activity.Duration.TotalSeconds, metricPoint.GetHistogramSum());
```

Comment:
is there test covering the scenario where instrumentTraces = false?, and it relies on its own timestamp to do duration?

Reply:
Tests were updated to to cover metrics + traces, only traces, only metrics and neither enabled. There is not a check for the Stopwatch based duration, however there is an assertion to ensure the histogram has a value.

Reply:
Got it.

_There is not a check for the Stopwatch based duration, however there is an assertion to ensure the histogram has a value_

Nonblocking: It'd be good to cover that to check the actual duration in a future improvement! (the test can setup its own stopwatch, and check that the duration calculated by instrumentaion is within a reasonable range of that.)